blazar/plugins/networks/network_plugin.py

```diff
@@ -634,6 +636,61 @@ def delete_network(self, network_id):
             # they have to rerun
             raise manager_ex.CantDeleteNetwork(network=network_id, msg=str(e))
 
+    def list_allocations(self, query, detail=False):
+        network_id_list = [n['id'] for n in db_api.network_list()]
+        options = self.get_query_options(query, QUERY_TYPE_ALLOCATION)
+        options['detail'] = detail
+
+        network_allocations = self.query_network_allocations(network_id_list,
+                                                             **options)
+        return [{"resource_id": network, "reservations": allocs}
+                for network, allocs in network_allocations.items()]
+
+    def get_allocations(self, network_id, query):
+        options = self.get_query_options(query, QUERY_TYPE_ALLOCATION)
+        network_allocations = self.query_network_allocations([network_id],
+                                                             **options)
+        allocs = network_allocations.get(network_id, [])
+        return {"resource_id": network_id, "reservations": allocs}
```

Comment:
Is this how the host allocations API works? I find this API a bit weird. It feels much simpler to just have one operation, `list`, which can have an optional network_id param. If that param is defined, it will return only allocations on that resource.

Reply:
Yes, but the code upstream for hosts is worse.
```
if host_id not in host_allocations:
            host_allocations = {host_id: []}
allocs = host_allocations[host_id]
```